### DISCUSSION TITLE
Typed data from list

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2504] [Q&A] [answered]
I'm receiving a List from an api that looks like `[[1 ,2, 'something', 32], [2 ,142, 'something else', 546]]` that is basically the result of an sql query. Is there a way convert the above in a drift data typed class? For example If I have a table `items`
that has the columns
`id, ref_id, value, price`
how can I create a list from the data that will end up looking like [ItemData, ItemData]?

drift is already doing this I assume but im not sure how and haven't found something in the documentation

**Comment by simolus3** [comment]
Drift does not do this exact thing internally, we can only map from a `Map` of column names to values to a row class. If you have it as a list without keys, there's no way to infer which value comes from which column. If you assume that the order matches the order of columns in the database, you could do

```dart
final row = [1, 2, 'something', 32];
final driftRow = YourTableRow.fromJson({
  for (var i = 0; i < row.length; i++) databaseClass.yourTable.$columns[i].name: row[i],
});
```

**Reply by BananaMasterz** [reply]
I tried this approach but `$columns` is snake case (easy to convert to camel case so not an issue) but then there's an issue I got stuck and came for help. If the api return `0` and that column in drift is bool, if I use `fromJson` drift will be expecting true or false but instead gets 0 or 1. 

If there's a way to translate the value making 0 false etc, that would be great. I assume something similar will happen with datetimes but not sure haven't tried yet. But not sure how/if that's possible

**Reply by simolus3** [reply, marked as answer]
In that case, using `databaseClass.yourTable.map` instead of `YourTableRow.fromJson` might help?

**Reply by BananaMasterz** [reply]
ah yeah that works great! Is there anything I should be aware of when doing this? Seems pretty straightforward

**Reply by simolus3** [reply]
> Is there anything I should be aware of when doing this?

Nothing comes to mind. If it works with your format, it'll continue to work.
